test/src/nyc-test.js

```diff
@@ -232,6 +248,39 @@ describe('nyc', function () {
       })
     })
 
+    describe('compile handlers for custom extensions are assigned', function () {
+      it('assigns a function to custom extensions', function () {
+        var nyc = new NYC({
+          cwd: process.cwd()
```

Comment:
The reason I didn't is because the tests will use package.json by default over the `opts` argument.  I patterned this after the `config.require` / `opts.require` that already exists.

I agree that it is not intuitive (I would expect `opts` to override `config`), but I figured the most important thing is consistency.  If you want, I can swap this around so that `opts` does override `config` for the `extension`.  I could submit a separate PR for the `require` config so that it behaves the same way.  Let me know your thoughts.